packages/framework/src/types/step.types.ts

```diff
@@ -64,7 +64,8 @@ export type ActionStep<
   /**
    * The controls for the step.
    */
-  T_Controls extends Record<string, unknown> = FromSchema<T_ControlSchema>,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  T_Controls extends Record<string, any> = FromSchema<T_ControlSchema>,
```

Comment:
Could you please clarify why these changes were needed?

We are preferring an approach of type strictness where assertions need to be made on the control properties types when schemas are not declared. If I'm not mistaken, this change will weaken that type strictness.

Reply:
as explained in the comment above. 
I can revert this change if you like, but this has to be looked on seperatly.

Reply:
I have reverted this after resolving the type inference issues.